source/KeyHandlers.js

```diff
@@ -428,6 +424,117 @@ var findPreviousTextOrNotEditable = function(root, node){
     return w.previousNode(notEditable)
 }
 
+
+/******************** NEIL JS VERSION OF ENTER *****************/
```

Comment:
we should read over the ENTER handler in detail and see if we can pull in desired behavior to our extracted function.